### ISSUE TITLE
Loading all Queue Drivers by default

### ISSUE THREAD

**Issue by ReactionJordan** [laravel/framework#10685]
When just using `sync` locally with queue the framework tries to load all the drivers. This led to a bit of confusion as the project I am on is far to small to need anything other then database or maybe redis.

The settings allowing multiple drivers is nice, however by default they should all be disabled unless they are being used. 

To clarify:
In http://laravel.com/docs/5.1/queues you can see it state
```
The following dependencies are needed for the listed queue drivers:

Amazon SQS: aws/aws-sdk-php ~3.0
Beanstalkd: pda/pheanstalk ~3.0
IronMQ: iron-io/iron_mq ~2.0
Redis: predis/predis ~1.0
```

However if you want to use Redis, it will require you to not only include `predis/predis ~1.0` in your `composer.json` but all the others as well, or you will experience fatal errors.

I have not tried with `Redis`, only `sync`, I used the above as example of an extension of this issue. The specific error I was receiving was that there was no Amazon SQS library available `Class 'Aws\Sqs\SqsClient' not found`.

However this can be fixed by commenting out all other drivers in `config/queue.php`, so it could just need a more descriptive comment in that file about how they get loaded.

**Comment by GrahamCampbell** [comment]
Can't replicate. You must be doing something yourself to cause that.

**Comment by andy1547** [comment]
I can confirm I had exactly the same issue.
This did not happen on Ubuntu 14/15 and Centos 5, but affected Centos 7 so it's an environment issue.
The PHP I'm using on Centos 7 is https://webtatic.com/packages/php56/.

As Graham suggested, commenting out the config in config/queue.php fixed this issue.

**Comment by ReactionJordan** [comment]
That makes sense, its easy to get around but confusing if it comes up. I have not checked for this in recent versions so not sure if anything was done differently in 5.2. However I have had more projects come up and have switched to redis queue and not seen this issue come up, so it could of been fixed in 5.2

Anyway I still think there is an advantage to a setting to tell laravel not use other drivers by default, opt-in rather then only being able to opt-out by removing the settings entirely.

Although there might be a little confusion on my original post, by "locally" I meant homestead so if it is indeed that it is due to ext-redis not being available then it would be a homestead issue, again I have not seen it recently so it could of only been one specific version of homestead (not sure what one it was now, long since updated).

So now I would say if there is any real fix for this, it would be 2 things. 1 is to do the opt-in settings, something that might be a bit annoying to current users, so probably not worth it. 2 is to make the documentation a little more clear on the requirements for the drivers as @andy1547 has said.
